src/Backend/Modules/Catalog/Engine/Model.php

```diff
@@ -130,6 +134,10 @@ public static function deleteCategory($id)
 			$db->delete('meta', 'id = ?', array($item['meta_id']));
 			$db->delete('catalog_categories', 'id = ?', array((int)$id));
 			$db->update('catalog_products', array('category_id' => null), 'category_id = ?', array((int)$id));
+
+			// delete extra and pages_blocks
+			BackendModel::deleteExtraById($item['extra_id']);
```

Comment:
The deleteExtraById function does not delete the page blocks by default. I think it is best to add a second parameter (boolean:true) so existing blocks get removed.